diff --git a/drivers/Kconfig b/drivers/Kconfig
index ab4d439..f0d8bfc 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -219,4 +219,6 @@ source "drivers/siox/Kconfig"
 
 source "drivers/slimbus/Kconfig"
 
+source "drivers/elphel/Kconfig"
+
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 578f469..8e73166 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -85,7 +85,7 @@ obj-$(CONFIG_SPI)		+= spi/
 obj-$(CONFIG_SPMI)		+= spmi/
 obj-$(CONFIG_HSI)		+= hsi/
 obj-$(CONFIG_SLIMBUS)		+= slimbus/
-obj-y				+= net/
+
 obj-$(CONFIG_ATM)		+= atm/
 obj-$(CONFIG_FUSION)		+= message/
 obj-y				+= firewire/
@@ -186,3 +186,9 @@ obj-$(CONFIG_MULTIPLEXER)	+= mux/
 obj-$(CONFIG_UNISYS_VISORBUS)	+= visorbus/
 obj-$(CONFIG_SIOX)		+= siox/
 obj-$(CONFIG_GNSS)		+= gnss/
+
+obj-$(CONFIG_ELPHEL393)		+= elphel/
+obj-$(CONFIG_ELPHELDRVONMICROZED)	+= elphel/
+obj-$(CONFIG_ELPHEL393_EXTERNAL)	+= elphel/
+
+obj-y				+= net/
