diff --git a/drivers/Kconfig b/drivers/Kconfig
index e1e2066..0970aa0 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -202,4 +202,6 @@ source "drivers/hwtracing/intel_th/Kconfig"
 
 source "drivers/fpga/Kconfig"
 
+source "drivers/elphel/Kconfig"
+
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 194d20b..caad273 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -81,7 +81,7 @@ obj-$(CONFIG_MTD)		+= mtd/
 obj-$(CONFIG_SPI)		+= spi/
 obj-$(CONFIG_SPMI)		+= spmi/
 obj-$(CONFIG_HSI)		+= hsi/
-obj-y				+= net/
+
 obj-$(CONFIG_ATM)		+= atm/
 obj-$(CONFIG_FUSION)		+= message/
 obj-y				+= firewire/
@@ -173,3 +173,9 @@ obj-$(CONFIG_STM)		+= hwtracing/stm/
 obj-$(CONFIG_ANDROID)		+= android/
 obj-$(CONFIG_NVMEM)		+= nvmem/
 obj-$(CONFIG_FPGA)		+= fpga/
+
+obj-$(CONFIG_ELPHEL393)		+= elphel/
+obj-$(CONFIG_ELPHELDRVONMICROZED)	+= elphel/
+obj-$(CONFIG_ELPHEL393_EXTERNAL)       += elphel/
+
+obj-y				+= net/
